lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2352,25 +2359,29 @@ balanceTransaction ctx genChange (ApiT wid) body = do
         -> W.ProtocolParameters
         -> Cardano.ProtocolParameters
         -> TransactionCtx
-        -> TransactionCtx
-    padFeeEstimation sealedTx pp pp' txCtx =
-        let
-            walletTx = decodeTx tl sealedTx
-            worseEstimate = calcMinimumCost tl pp txCtx skeleton
-            skeleton = SelectionSkeleton
+        -> Either ErrUpdateSealedTx TransactionCtx
+    padFeeEstimation sealedTx' pp pp' txCtx = do
+        -- Ensure `evaluateMinimumFee` always will overestimate even though the
+        -- execution units are not known yet. Calling `updateTx` like this will:
+        -- 1. make the CBOR encoding of the execution units as large as possible
+        -- 2. add the script integrity hash
+        let maxUnits = pp ^. (#txParameters . #getMaxExecutionUnits)
```

Comment:
Note that the max units are already part of the context as `txPlutusScriptExecutionCost`, so it'd be better to pull from there and avoid duplicating this.